test/powershell/Modules/Microsoft.PowerShell.Utility/xml.tests.ps1

```diff
@@ -55,19 +55,31 @@
         }
 
         It "non filesystem path using <parameter>" -TestCases @(
-            @{parameter="literalPath"},
-            @{parameter="path"}
+            @{parameter="literalPath"; expectedError='ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand'},
+            @{parameter="path"; expectedError='ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand'}
         ) {
-            param($parameter)
-            $__data = "abcdefg"
-            $params = @{$parameter="variable:__data"}
-            { Select-XML @params "Root" | Should BeErrorId 'ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand' }
+            param($parameter, $expectedError)
+            try
+            {
+                $env:xmltestfile = '<a><b>'
+                $file = 'env:xmltestfile'
+                $params = @{$parameter=$file}
+                $err = $null
+                Select-XML @params "Root" -ErrorVariable err
+                $err.FullyQualifiedErrorId | Should Be $expectedError
+            }
+            finally
+            {
+                Remove-Item -Path 'env:xmltestfile'
+            }
         }
 
         It "Invalid xml file" {
             $testfile = "$testdrive/test.xml"
             Set-Content -Path $testfile -Value "<a><b>"
-            { Select-Xml -Path $testfile -XPath foo | ShouldBeErrorId 'ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand' }
+            $err = $null
+            Select-Xml -Path $testfile -XPath foo -ErrorVariable err
+            $err.FullyQualifiedErrorId | Should Be 'ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand'
```

Comment:
Use `ShouldBeErrorId`

Reply:
Select-Xml does not throw a terminating error.